### PR TITLE
[PDR-805] 'Prefer not to answer' should not be RBR.

### PR DESCRIPTION
**Pull request by robabram**
## Description of changes/additions
Due to changes in the Program Data Glossary, when a participant answers with 'PMI_PreferNotToAnswer', these should not be counted as RBR.

## Tests
- [x] unit tests

### CODE REVIEW

File: rdr_service/resource/calculators/participant_ubr.py [hunk 1]
```diff
@@ -53,8 +54,7 @@ def ubr_sexual_orientation(answer: (str, None)):
         # they did not answer the expected SexualOrientation_None response to the parent TheBasics_SexualOrientation
         # question first.  This is not  consistent with the documented survey branching logic.  Therefore, UBR/RBR
         # calculations are still based only on the response to the TheBasics_SexualOrientation question alone
-
-        if answer is None or answer == 'PMI_Skip':
+        if answer in [None, 'PMI_Skip', 'PMI_PreferNotToAnswer']:
             return UBRValueEnum.NotAnswer_Skip
 
         if answer not in ['SexualOrientation_Straight', 'PMI_PreferNotToAnswer']:
```
Comment (pegbertsch) [inline rdr_service/resource/calculators/participant_ubr.py]: Can we just change this to `if answer != 'SexualOrientation_Straight` now, for clarity?

File: rdr_service/resource/calculators/participant_ubr.py [hunk 2]
```diff
@@ -93,14 +93,13 @@ def ubr_gender_identity(birth_sex, gender_ident, gender_ident_closer):
```
Comment (pegbertsch) [inline rdr_service/resource/calculators/participant_ubr.py:92]: Above on line 86/87:  Do we also need to account for potential unexpected/non-conformant response data where they could have skipped (`PMI_Skip`) the `gender_ident` question but provided a non-skip answer to `gender_ident_closer`?

File: rdr_service/resource/calculators/participant_ubr.py [hunk 3]
```diff
@@ -132,13 +131,14 @@ def ubr_ethnicity(answers):
         """
         if answers is None:
             return UBRValueEnum.NotAnswer_Skip
+
         # Note: assume UBR by default and check for exclusion criteria.
         answers = answers.split(',')
         if len(answers) == 1:
```
Comment (pegbertsch) [inline rdr_service/resource/calculators/participant_ubr.py:136]: Do we need to confirm there is no unexpected/non-conformant response data ingested by the RDR where there are multiple answer codes but one of them is `PMI_Skip` or `PMI_PreferNotToAnswer`?

**Review by pegbertsch — CHANGES_REQUESTED** [review]
Asking for the update in `ubr_sexual_orientation()` line 60, since if `PMI_PreferNotToAnswer` was the answer code passed in, we already would have returned a value on line 58

File: rdr_service/resource/calculators/participant_ubr.py (hunk 1 above)
Comment (robabram) [reply rdr_service/resource/calculators/participant_ubr.py]: Yes, I'll make that change.

File: rdr_service/resource/calculators/participant_ubr.py (hunk 3 above)
Comment (robabram) [reply rdr_service/resource/calculators/participant_ubr.py:136]: I looked at the possible answers for 'Race_WhatRaceEthnicity' and there are no cases where a Skip or Prefer Not To Answer are mixed with other answers.  I think this logic is fine in this method.

File: rdr_service/resource/calculators/participant_ubr.py (hunk 2 above)
Comment (robabram) [reply rdr_service/resource/calculators/participant_ubr.py:92]: I checked, there are no cases where Gender_GenderIdentity has PMI_Skip and Gender_CloserGenderDescription has a non-skip answer.

**Review by pegbertsch — APPROVED** [review]

**Review by dtharpeuno — APPROVED** [review]

**Pull request merged**